src/cli/onefuzz/status/cache.py

```diff
@@ -106,194 +139,195 @@ def __init__(
         self.endpoint: str = endpoint
         self.last_update = datetime.now()
 
-    def add_container(self, name: str, ignore_date: bool = False) -> None:
+    def add_container(self, name: Container) -> None:
         if name in self.files:
             return
         try:
             files = self.onefuzz.containers.files.list(name)
         except Exception:
             return
 
-        self.add_files(name, set(files.files), ignore_date=ignore_date)
+        self.add_files_set(name, set(files.files))
+
+    def add_message(self, message: EventMessage) -> None:
+        events = {
+            EventPoolCreated: lambda x: self.pool_created(x),
+            EventPoolDeleted: lambda x: self.pool_deleted(x),
+            EventTaskCreated: lambda x: self.task_created(x),
+            EventTaskStopped: lambda x: self.task_stopped(x),
+            EventTaskFailed: lambda x: self.task_stopped(x),
+            EventTaskStateUpdated: lambda x: self.task_state_updated(x),
+            EventJobCreated: lambda x: self.job_created(x),
+            EventJobStopped: lambda x: self.job_stopped(x),
+            EventNodeStateUpdated: lambda x: self.node_state_updated(x),
+            EventNodeCreated: lambda x: self.node_created(x),
+            EventNodeDeleted: lambda x: self.node_deleted(x),
+            EventCrashReported: lambda x: self.file_added(x),
+            EventFileAdded: lambda x: self.file_added(x),
+        }
+
+        for event_cls in events:
+            if isinstance(message.event, event_cls):
+                events[event_cls](message.event)
 
-    def add_message(self, name: str, message: Dict[str, Any]) -> None:
         self.last_update = datetime.now()
-        data: Dict[str, Union[int, str]] = {}
-        for (k, v) in message.items():
-            if k in ["task_id", "job_id", "pool_id", "scaleset_id", "machine_id"]:
-                k = k.replace("_id", "")
-                data[k] = str(v)[:8]
-            elif isinstance(v, (int, str)):
-                data[k] = v
-
-        as_str = fmt(data)
-        messages = [x for x in self.messages if (x[1:] != [name, as_str])][-99:]
-        messages += [(datetime.now(), name, as_str)]
-
+        messages = [x for x in self.messages][-99:]
+        messages += [
+            (datetime.now(), message.event_type, message.event.json(exclude_none=True))
```

Comment:
This isn't new, but should we be explicitly specifying the TZ as UTC in all our calls to `now()`?

Reply:
In general, yes.  However, in the case of displaying to users, it made sense to use `now()` here.

This only displays the local timestamp of when the message was received.